File: scrudbeans-integration-tests-java/src/main/java/myjavapackage/service/OrderLineService.java
package myjavapackage.service;


import com.github.manosbatsis.scrudbeans.repository.ModelRepository;
import com.github.manosbatsis.scrudbeans.service.PersistableModelService;
import com.github.manosbatsis.scrudbeans.service.AbstractPersistableModelServiceImpl;
import myjavapackage.model.OrderLine;
import myjavapackage.model.Product;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class OrderLineService extends AbstractPersistableModelServiceImpl<OrderLine, String, ModelRepository<OrderLine, String>>
		implements PersistableModelService<OrderLine, String> {

	private ModelRepository<Product, String> productRepository;

	/**
	 * {@inheritDoc}
	 * @param resource
	 */
	@Override
	public OrderLine create(OrderLine resource) {
		return super.create(resource);
	}

	@Autowired
	public void setProductRepository(ModelRepository<Product, String> productRepository) {
		this.productRepository = productRepository;
	}
}

File: scrudbeans-integration-tests-kotlin/src/main/kotlin/mykotlinpackage/service/OrderLineService.kt
package mykotlinpackage.service

import com.github.manosbatsis.scrudbeans.service.AbstractJpaEntityProjectorService
import jakarta.persistence.EntityManager
import mykotlinpackage.model.OrderLine
import mykotlinpackage.model.OrderLineIdentifierAdapter
import mykotlinpackage.repository.OrderLineRepository
import org.springframework.context.annotation.Lazy
import org.springframework.stereotype.Service
import java.util.UUID

@Service
class OrderLineService(
    orderLineRepository: OrderLineRepository,
    entityManager: EntityManager,
    @Lazy identifierAdapter: OrderLineIdentifierAdapter,
) : AbstractJpaEntityProjectorService<OrderLine, UUID, OrderLineRepository>(
    orderLineRepository,
    entityManager,
    identifierAdapter,
)
